packages/editor/src/components/post-publish-panel/index.js

```diff
@@ -60,12 +60,14 @@ export class PostPublishPanel extends Component {
 			PrePublishExtension,
 			...additionalProps
 		} = this.props;
-		const isPublishedOrScheduled = isPublished || ( isScheduled && isBeingScheduled );
 		const propsForPanel = omit( additionalProps, [ 'hasPublishAction', 'isDirty' ] );
+		const isPublishedOrScheduled = isPublished || ( isScheduled && isBeingScheduled );
+		const isPrePublish = ! isPublishedOrScheduled && ! isSaving;
```

Comment:
I'm wondering if this `isSaving` check causes re-renders if an autosave fires after the panel is open? I saw some weird re-renders on slower connections (I'm working from the pub right now) and I wonder if a late auto-save after the publish panel is opened causes this to close the panel and then open it again?

Reply:
Not sure about how the autosave mechanism works, but I've just checked the editor store code and `isSaving` (that holds `state.saving.request`) only changes when the content has been edited. So, the spinner will be shown instead of the pre-publish panel when: 

* the post contents are changed while the panel is open
* the panel was opened before the auto-save kicked in

Note that I couldn't reproduce this with the post-publish panel open. 

For reference, before https://github.com/WordPress/gutenberg/pull/11013 this wasn't probably happening because we used some internal state (`submitted`) to keep track of this, which had introduced a bug in scheduling posts.

Happy to revisit this in a separate PR if needed. Just assign it to me!

Reply:
Forgot to mention: I've noticed that focus is retained at the component that had it after spinner is rendered.